dask/bag/core.py

```diff
@@ -297,19 +352,37 @@ def remove(self, predicate):
                    for i in range(self.npartitions))
         return type(self)(merge(self.dask, dsk), name, self.npartitions)
 
-    def map_partitions(self, func):
+    def map_partitions(self, func, **kwargs):
         """ Apply function to every partition within collection
 
         Note that this requires you to understand how dask.bag partitions your
         data and so is somewhat internal.
 
         >>> b.map_partitions(myfunc)  # doctest: +SKIP
+
+        Any **kwargs are passed through to func along with the (maybe unpacked)
+        item argument(s).  Values in **kwargs must either be a non-dask python
+        literal or a dask.bag.Item.
+
+        If a dask.Bag.Item is used its computed value is passed to the map
+        function.  This means that whatever computations are implied by any
+        Item values block any of the map work.  Example:
+
+        >>> import dask.bag as db
+        >>> b = db.from_sequence(range(100), npartitions=10)
+        >>> b.map_partitions(
+        ...     lambda X, total=0: [float(x) / total for x in X],
+        ...     total=b.sum()).sum().compute()
+        1.0
         """
         name = 'map-partitions-{0}-{1}'.format(funcname(func),
-                                               tokenize(self, func))
-        dsk = dict(((name, i), (func, (self.name, i)))
+                                               tokenize(self, func, kwargs))
+        dsk = self.dask.copy()
+        dsk, kw_pairs = merge_kwargs(dsk, kwargs)
+        dsk.update(((name, i),
+                    (apply, func, [(self.name, i)], (dict, kw_pairs)))
```

Comment:
Would using just `apply` work here instead?  Generally it's nice to keep information in the dask graph rather than hidden away in `partial`s or closures.  This makes debugging much easier.  It is also one less function that dask developers need to keep track of in the future.